components/shared/onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,320 @@
+        changeStep();
+      } else if (currentStep === "4") {
+        await updateUser({
+          updateData: {
+            newProjects,
+            startTime,
+            endTime,
+          },
+        });
+        router.push("/");
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const renderForm = () => {
+    switch (currentStep) {
+      case "1":
+        return (
+          <OnboardingOne
+            user={user}
+            step={step}
+            register={register}
+            errors={errors}
+          />
+        );
+      case "2":
+        return (
+          <OnboardingTwo
+            step={step}
+            goalFields={goalFields}
+            register={register}
+            appendGoal={appendGoal}
+            removeGoal={removeGoal}
+          />
+        );
+      case "3":
+        return (
+          <OnboardingThree
+            register={register}
+            step={step}
+            experiences={experienceFields}
+            errors={errors}
+            appendExperience={appendExperience}
+            removeExperience={removeExperience}
+          />
+        );
+      case "4":
+        return (
+          <OnboardingFour
+            register={register}
+            control={control}
+            selectedFrom={selectedFrom!}
+            selectedTo={selectedTo!}
+            handleDaySelectFrom={handleDaySelectFrom}
+            handleDaySelectTo={handleDaySelectTo}
+            errors={errors}
+            step={step}
+          />
+        );
+      default:
+        return (
+          <OnboardingOne
+            user={user}
+            step={step}
+            register={register}
+            errors={errors}
+          />
+        );
+    }
+  };
+  useEffect(() => {
+    switch (step) {
+      case "1":
+        setCurrentStep("1");
+        break;
+      case "2":
+        setCurrentStep("2");
+        break;
+      case "3":
+        setCurrentStep("3");
+        break;
+      case "4":
+        setCurrentStep("4");
+        break;
+      default:
+        setCurrentStep("1");
+        break;
+    }
+  }, [step]);
+
+  return (
+    <>
+      <section className="relative w-full">
+        <div className="absolute top-[50%] h-1 w-full bg-gray-400/30"></div>
+        <Progress
+          className="absolute top-[25%]"
+          value={(Number(currentStep) - 1) * 33}
+          max={4}
+        />
+
+        <OnboardingTick step={parseInt(currentStep)} />
+      </section>
+      {currentStep === "1" && (
+        <div className="box-border justify-between border-white-500 p-4">
+          {" "}
+          <UploadPhoto image={user.image} />
+        </div>
+      )}
+      <form
+        className="flex-0 w-full overflow-y-hidden"
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        {currentStep && renderForm()}
+
+        {currentStep !== "4" ? (
+          <CustomButton
+            className="pb-4"
+            buttonType="primary"
+            onClick={changeStep}
```

Comment:
```suggestion
          <CustomButton
            className="pb-4"
            buttonType="primary"
            type="button"
            onClick={changeStep}
```
We need to make this a button so it doesn't always try and submit the form.

https://www.w3schools.com/html/html_forms.asp